### ISSUE TITLE
Error in table creation Index column size too large. The maximum column size is 767 bytes.

### ISSUE THREAD

**Issue by Csudam** [ThemeFuse/Unyson#1629]
**Index column size too large. The maximum column size is 767 bytes.** on plugin activation with WordPress version 4.2 and above since it is using the character set utf8mb4 this issue is not occurring with the character set utf8. The issue is occurring in following create table syntax.
CREATE TABLE wp_fw_termmeta (
				meta_id bigint(20) unsigned NOT NULL auto_increment,
				fw_term_id bigint(20) unsigned NOT NULL default '0',
				meta_key varchar(255) default NULL,
				meta_value longtext,
				PRIMARY KEY	(meta_id),
				KEY fw_term_id (fw_term_id),
				KEY meta_key (meta_key)
				) DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;

**Comment by Csudam** [comment]
Found that the issue is occurring due to the character set utf8mb4, if I change this to utf8 it is working perfectly.Since from WordPress v 4.2, it has switched to the character set utf8mb4 from utf8.

**Comment by moldcraft** [comment]
Not an answer, just a note for myself: WordPress `wp_termmeta` table query is:

```sql
CREATE TABLE `wp_termmeta` (
  `meta_id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
  `term_id` bigint(20) unsigned NOT NULL DEFAULT '0',
  `meta_key` varchar(255) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
  `meta_value` longtext COLLATE utf8mb4_unicode_ci,
  PRIMARY KEY (`meta_id`),
  KEY `term_id` (`term_id`),
  KEY `meta_key` (`meta_key`(191))
) ENGINE=InnoDB AUTO_INCREMENT=7 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci
```

Note `KEY `meta_key` (`meta_key`(191))`

**Comment by Csudam** [comment]
I am getting this error in fresh installation of WordPress and unison. If the tables in the database are already created then this error will not occur because the table already exists in the database so it will not be going to create a table.

**Comment by moldcraft** [comment]
Please replace [this line](https://github.com/ThemeFuse/Unyson/blob/v2.5.5/framework/includes/term-meta/function_fw_term_meta_setup_blog.php#L32)  with

```sql
KEY `meta_key` (`meta_key`(191))
```

and tell us if it helps

**Comment by Csudam** [comment]
Yes, this works also I found the reason behind this it is character set utf8mb4 used in create table syntax. if I change it to the utf8 it will work fine and I don't need to specify the size in  KEY `meta_key` (`meta_key`(191)).Using utf8 we can create the index on column `meta_key`  varchar(255).
I have changed it and it is working now.
But the problem is WordPress has  changed the default character set to utf8mb4 using this character set the key size will exceed the 1020 bytes and which exceeds the limit of index column. so it should be fixed in unyson plugin.
It can be fixed by either reducing the column size or by changing the character set to utf8.
